stl/inc/ranges

```diff
@@ -22,15 +25,461 @@ _STL_DISABLE_CLANG_WARNINGS
 
 _STD_BEGIN
 namespace ranges {
-    // Much machinery defined in <xutility>
+    // MUCH machinery defined in <xutility>
 
     // clang-format off
     // CONCEPT ranges::viewable_range
     template <class _Rng>
     concept viewable_range = range<_Rng>
         && (borrowed_range<_Rng> || view<remove_cvref_t<_Rng>>);
+
+    // CLASS TEMPLATE ranges::_Semiregular_box
+    // clang-format off
+    template <class _Ty>
+    concept _Copy_constructible_object = copy_constructible<_Ty> && is_object_v<_Ty>;
+    // clang-format on
+
+#if 0 // TRANSITION, VSO-1174090
+    template <_Copy_constructible_object _Ty>
+    class _Semiregular_box {
+    public:
+        constexpr _Semiregular_box() noexcept : _Dummy{}, _Engaged{false} {}
+        constexpr _Semiregular_box() noexcept(is_nothrow_default_constructible_v<_Ty>) requires constructible_from<_Ty>

```

Comment:
The [standard ](http://eel.is/c++draft/range.semi.wrap#1.2) talks about `default_­initializable` which is more than just `constructible_from<_Ty>`

Reply:
Yes it does. I'm relaxing this requirement ever-so-slightly from the Standard's _`default-initializable`_ to `constructible_from<_Ty>` (which is effectively "`_Ty` can be value-initialized") since we value-init the stored object with this type. We weren't fully cognizant of the distinction between `default_initializable<T>` and `constructible_from<T>` when this wording was written. I ought to change the Standard wording, but it's hard to motivate given that I don't know of any types that model `copyable<T> && constructible_from<T>` but not `semiregular<T>` (which is `copyable<T> && default-initializable<T>`).

Reply:
I've reminded myself that the primary goal of constraints is not to be minimal, and changed these occurrences of `constructible_from<T>` to `default_initializable<T>`.

Reply:
Thanks for following up in thid